src/optimizer/plan_nodes/logical_aggregate.rs

```diff
@@ -69,13 +69,16 @@ impl_plan_tree_node_for_unary!(LogicalAggregate);
 impl PlanNode for LogicalAggregate {
     fn schema(&self) -> Vec<ColumnDesc> {
         let child_schema = self.child.schema();
-        let mut input_refs = vec![];
         self.group_keys
             .iter()
-            .for_each(|expr| expr.resolve_input_ref(&mut input_refs));
-        input_refs
-            .iter()
-            .map(|expr| child_schema[expr.index].clone())
+            .enumerate()
+            .map(|(index, expr)| {
+                ColumnDesc::new(
+                    child_schema[index].datatype().clone(),
```

Comment:
still doesn't seem correct? we should use expression's return type instead of child schema's type.